app/subjects/index.cjsx

```diff
@@ -0,0 +1,81 @@
+React = require 'react'
+apiClient = require '../api/client'
+authClient = require '../api/auth'
+talkClient = require '../api/talk'
+getSubjectLocation = require '../lib/get-subject-location'
+FavoritesButton = require '../collections/favorites-button'
+ChangeListener = require '../components/change-listener'
+PromiseRenderer = require '../components/promise-renderer'
+NewDiscussionForm = require '../talk/discussion-new-form'
+{Navigation} = require 'react-router'
+
+module?.exports = React.createClass
+  displayName: 'Subject'
+  mixins: [Navigation]
+
+  getInitialState: ->
+    subject: null
+
+  componentWillMount: ->
+    @setSubject()
+
+  setSubject: ->
+    subjectId = @props.params?.id.toString()
+    apiClient.type('subjects').get(subjectId)
+      .then (subject) =>
+        @setState {subject}
+
+  comment: (data, i) ->
+    <div key={data.id} className="talk-module">
+      <strong>{data.user_display_name}</strong>
+      <br />
+      <span>{data.body}</span>
+    </div>
+
+  onCreateDiscussion: (discussion) ->
+    projectId = discussion.section.split('-')[0] # string
+    apiClient.type('projects').get(projectId).then (project) =>
+      project.get('owner').then (owner) =>
+        @transitionTo('project-talk-discussion', {owner: owner.slug, name: project.slug, board: discussion.board_id, discussion: discussion.id})
+
+  render: ->
+    {subject} = @state
+
+    <div className="subject talk content-container">
+      {if subject
+        <section>
+          <h1>Subject {subject.id}</h1>
+
+          <img src={getSubjectLocation(subject).src} />
```

Comment:
Maybe use components/subject-viewer here to keep subject rendering consistent and to make sure eventual non-image subjects are rendered correctly.